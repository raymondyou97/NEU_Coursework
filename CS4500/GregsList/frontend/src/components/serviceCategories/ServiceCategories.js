import React from 'react';

import '../../styles/serviceCategories.css';

class ServiceCategories extends React.Component {
  componentDidMount() {
    this.props.fetchServiceCategories();
  }

  updateTitle = e => {
    this.props.updateTitle(e.target.value);
  };

  updateDescription = e => {
    this.props.updateDescription(e.target.value);
  };

  deleteServiceCategory = id => {
    this.props.deleteServiceCategory(id);
  };

  editServiceCategory = (id, title, description) => {
    this.props.updateId(id);
    this.props.updateTitle(title);
    this.props.updateDescription(description);
  };

  render() {
    const { serviceCategory } = this.props;

    return (
      <div>
        <h3 className="service-category-title">Service Categories</h3>
        <table className="service-category-table">
          <tbody className="service-category-table-body">
            <tr className="service-category-columns">
              <td>Title</td>
              <td>Description</td>
              <td></td>
            </tr>
            <tr className="service-category-inputs">
              <td>
                <input
                  className="service-category-title-input"
                  type="text"
                  placeholder="Title"
                  value={serviceCategory.title}
                  onChange={this.updateTitle}
                />
              </td>
              <td>
                <input
                  className="service-category-description-input"
                  type="text"
                  placeholder="Description"
                  value={serviceCategory.description}
                  onChange={this.updateDescription}
                />
              </td>
              <td className="service-category-input-buttons">
                <button
                  className="service-category-add-button"
                  onClick={this.props.createServiceCategory}
                >
                  Add
                </button>
                <button
                  className="service-category-update-button"
                  onClick={this.props.updateServiceCategory}
                >
                  Update
                </button>
              </td>
            </tr>
            {this.props.serviceCategories.map(serviceCategory => {
              const { id, title, description } = serviceCategory;
              return (
                <tr className="service-category-details" key={id}>
                  <td onClick={() => this.props.showDetails(id)}>{title}</td>
                  <td onClick={() => this.props.showDetails(id)}>
                    {description}
                  </td>
                  <td className="service-category-buttons">
                    <button
                      className="service-category-delete-button"
                      onClick={() => this.deleteServiceCategory(id)}
                    >
                      Delete
                    </button>
                    <button
                      className="service-category-edit-button"
                      onClick={() =>
                        this.editServiceCategory(id, title, description)
                      }
                    >
                      Edit
                    </button>
                  </td>
                </tr>
              );
            })}
          </tbody>
        </table>
      </div>
    );
  }
}

export default ServiceCategories;
