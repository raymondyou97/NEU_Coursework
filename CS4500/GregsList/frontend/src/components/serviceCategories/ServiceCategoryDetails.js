import React, { Component } from 'react';
import ServiceCategoryService from '../../services/ServiceCategoryService';

import '../../styles/serviceCategoryDetails.css';

class ServiceCategoryDetails extends Component {
  constructor(props) {
    super(props);
    this.serviceCategoryService = ServiceCategoryService.getInstance();
    this.state = {
      id: this.props.match.params.id,
      serviceCategories: [],
      title: '',
      description: '',
      popularity: '',
    };
  }

  componentDidMount() {
    this.serviceCategoryService
      .findAllServiceCategories()
      .then(serviceCategories => {
        this.setState({
          serviceCategories: serviceCategories,
        });
      });
    this.selectServiceCategory(this.state.id);
  }

  updateTitle = e => {
    this.setState({ title: e.target.value });
  };

  updateDescription = e => {
    this.setState({ description: e.target.value });
  };

  updatePopularity = e => {
    this.setState({ popularity: e.target.value });
  };

  selectServiceCategory = id => {
    this.serviceCategoryService
      .findServiceCategoryById(id)
      .then(serviceCategory => {
        this.props.history.push('/admin/service-categories/' + id);
        this.setState({
          title: serviceCategory.title,
          description: serviceCategory.description,
          popularity: serviceCategory.popularity,
        });
      });
  };

  backToServiceCategories = () => {
    this.props.history.push('/admin/service-categories/');
  };

  createUser = () => {
    const { title, description, popularity } = this.state;

    if (title && description && popularity) {
      this.serviceCategoryService
        .createServiceCategory({
          title,
          description,
          popularity,
        })
        .then(response => {
          this.props.history.push('/admin/service-categories/');
        });
    }
  };

  deleteServiceCategory = id => {
    this.serviceCategoryService.deleteServiceCategory(id).then(response => {
      this.props.history.push('/admin/service-categories/');
    });
  };

  updateServiceCategory = () => {
    const { id, title, description, popularity } = this.state;
    let serviceCategoryData = {
      id,
      title,
      description,
      popularity,
    };
    this.serviceCategoryService
      .updateServiceCategory(id, serviceCategoryData)
      .then(response => {
        alert('Service category updated!');
      });
  };

  render() {
    return (
      <div className="service-category-detail-container">
        <h3 className="service-category-detail-sc-title">{this.state.title}</h3>
        <span className="service-category-detail-row">
          <span className="service-category-detail-title">
            Service Categories
          </span>
          <span className="service-category-detail-input">
            <select
              value={this.state.id}
              onChange={e => this.selectServiceCategory(e.target.value)}
              className="form-control"
            >
              {this.state.serviceCategories.map(serviceCategory => {
                return (
                  <option value={serviceCategory.id} key={serviceCategory.id}>
                    {serviceCategory.title}
                  </option>
                );
              })}
            </select>
          </span>
        </span>
        <span className="service-category-detail-row">
          <span className="service-category-detail-title">Title</span>
          <span className="service-category-detail-input">
            <input
              onChange={this.updateTitle}
              className="form-control"
              value={this.state.title}
            />
          </span>
        </span>
        <span className="service-category-detail-row">
          <span className="service-category-detail-title">Description</span>
          <span className="service-category-detail-input">
            <input
              onChange={this.updateDescription}
              className="form-control"
              value={this.state.description ? this.state.description : ''}
            />
          </span>
        </span>
        <span className="service-category-detail-row">
          <span className="service-category-detail-title">Popularity</span>
          <span className="service-category-detail-input">
            <input
              onChange={this.updatePopularity}
              className="form-control"
              value={this.state.popularity ? this.state.popularity : ''}
            />
          </span>
        </span>
        <span className="service-category-detail-buttons">
          <button
            className="service-category-detail-cancel"
            onClick={this.backToServiceCategories}
          >
            Cancel
          </button>
          <button
            className="service-category-detail-delete"
            onClick={() => this.deleteServiceCategory(this.state.id)}
          >
            Delete
          </button>
          <button
            className="service-category-detail-create"
            onClick={this.createServiceCategory}
          >
            Create
          </button>
          <button
            className="service-category-detail-update"
            onClick={this.updateServiceCategory}
          >
            Update
          </button>
        </span>
      </div>
    );
  }
}

export default ServiceCategoryDetails;
