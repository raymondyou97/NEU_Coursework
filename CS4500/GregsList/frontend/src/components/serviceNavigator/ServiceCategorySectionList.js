import React from 'react';
import ServiceCategorySection from './ServiceCategorySection';

const ServiceCategorySectionList = ({ serviceCategories }) => (
  <ul>
    {serviceCategories.map(serviceCategory => (
      <li key={serviceCategory.id} style={{ listStyle: 'none' }}>
        <div>
          <ServiceCategorySection serviceCategory={serviceCategory} />
        </div>
      </li>
    ))}
  </ul>
);

export default ServiceCategorySectionList;
