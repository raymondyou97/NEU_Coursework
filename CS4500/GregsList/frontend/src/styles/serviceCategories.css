td {
  padding: 16px 8px;
  box-sizing: border-box;
  font-family: 'Times New Roman';
}

tr {
  padding: 0px 16px;
}

button {
  height: 25px;
  color: white;
  border-radius: 3px;
  border: 0;
}

.service-category-title {
  margin-bottom: 0;
}

.service-category-table {
  width: 67%;
  margin: 16px auto;
  display: flex;
  flex: 0 1;
  border: 1px solid #556270;
  border-radius: 3px;
}

.service-category-table-body {
  display: inline-block;
  width: 100%;
}

.service-category-details {
  display: flex;
  flex: 1 1;
}

.service-category-columns {
  border-bottom: 1px solid #333;
  font-weight: bold;
  display: flex;
  flex: 1 1;
}

.service-category-columns td {
  display: flex;
  flex: 1 0;
  font-size: 20px;
}

.service-category-details td {
  display: flex;
  box-sizing: border-box;
  flex: 1 0;
}

.service-category-detail-buttons {
  display: flex;
  flex: 1 1;
}

.service-category-buttons button {
  display: flex;
  flex: 1 0;
}

.service-category-buttons button:first-child {
  margin-right: 4px;
}

.service-category-inputs {
  display: flex;
  flex: 1 1;
}

.service-category-inputs td {
  display: flex;
  flex: 1 0;
}

.service-category-inputs td input {
  width: 100%;
  height: 19px;
  border-radius: 3px;
  border: 1px solid #d5ded9;
}

.service-category-input-buttons button {
  display: flex;
  flex: 1 1;
}

.service-category-input-buttons button:first-child {
  margin-right: 4px;
}

.service-category-add-button,
.service-category-edit-button {
  background-color: #218838;
}

.service-category-update-button {
  background-color: #007bff;
}

.service-category-delete-button {
  background-color: #dc3545;
}
