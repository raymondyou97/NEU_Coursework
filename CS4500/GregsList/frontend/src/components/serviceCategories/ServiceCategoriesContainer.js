'use es6';

import React, { PureComponent } from 'react';

import ServiceCategoryService from '../../services/ServiceCategoryService';
import ServiceCategories from './ServiceCategories';

class ServiceCategoriesContainer extends PureComponent {
  constructor(props) {
    super(props);
    this.serviceCategoryService = ServiceCategoryService.getInstance();
    this.state = {
      id: -1,
      title: '',
      description: '',
      serviceCategories: [],
    };
  }

  updateTitle = title => {
    this.setState({ title });
  };

  updateDescription = description => {
    this.setState({ description });
  };

  updateId = id => {
    this.setState({ id });
  };

  fetchServiceCategories = () => {
    this.serviceCategoryService
      .findAllServiceCategories()
      .then(serviceCategories => {
        this.setState({
          serviceCategories: serviceCategories,
        });
      });
  };

  createServiceCategory = () => {
    const { title, description, serviceCategories } = this.state;

    if (title && description) {
      let titleConflict = false;
      this.state.serviceCategories.forEach(category => {
        if (title === category.title) {
          titleConflict = true;
        }
      });
      if (titleConflict) {
        alert(
          "Can't create service category with the same title as another service category"
        );
      } else {
        this.serviceCategoryService
          .createServiceCategory({
            title,
            description,
          })
          .then(response => {
            this.setState({
              id: -1,
              title: '',
              description: '',
              serviceCategories: serviceCategories.concat(response),
            });
          });
      }
    }
  };

  deleteServiceCategory = id => {
    this.serviceCategoryService.deleteServiceCategory(id);
    const newServiceCategories = this.state.serviceCategories.filter(sc => {
      return sc.id !== id;
    });
    this.setState({
      serviceCategories: newServiceCategories,
    });
  };

  // editServiceCategory = (id, title, description) => {
  //   this.setState({
  //     id,
  //     title,
  //     description,
  //   })
  // };

  updateServiceCategory = () => {
    const { id, title, description } = this.state;
    if (id !== -1 && title && description) {
      let titleConflict = false;
      this.state.serviceCategories.forEach(category => {
        if (title === category.title && id !== category.id) {
          titleConflict = true;
        }
      });
      if (titleConflict) {
        alert(
          "Can't create service category with the same title as another service category"
        );
      } else {
        this.serviceCategoryService.updateServiceCategory(id, {
          id,
          title,
          description,
        });
        let newServiceCategories = this.state.serviceCategories;
        newServiceCategories.forEach((val, ind) => {
          if (val.id === id) {
            newServiceCategories[ind] = {
              id,
              title,
              description,
            };
          }
        });
        this.setState({
          id: -1,
          title: '',
          description: '',
        });
      }
    }
  };

  showDetails = id => {
    this.props.history.push(`/admin/service-categories/${id}`);
  };

  render() {
    const { title, description } = this.state;

    const serviceCategory = {
      title,
      description,
    };

    return (
      <ServiceCategories
        fetchServiceCategories={this.fetchServiceCategories}
        serviceCategory={serviceCategory}
        updateId={this.updateId}
        serviceCategories={this.state.serviceCategories}
        updateTitle={this.updateTitle}
        updateDescription={this.updateDescription}
        createServiceCategory={this.createServiceCategory}
        showDetails={this.showDetails}
        deleteServiceCategory={this.deleteServiceCategory}
        updateServiceCategory={this.updateServiceCategory}
      />
    );
  }
}

export default ServiceCategoriesContainer;
