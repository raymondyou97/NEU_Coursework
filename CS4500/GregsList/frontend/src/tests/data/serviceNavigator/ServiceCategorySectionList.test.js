import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import ServiceCategorySectionList from '../../../components/serviceNavigator/ServiceCategorySectionList';
import serviceCategories from './service-categories.mock.json';

test('Service Category Section List renders correct number of items', () => {
  const component = shallow(
    <ServiceCategorySectionList serviceCategories={serviceCategories} />
  );
  expect(component.find('li').length).toEqual(serviceCategories.length);

  const firstRow = component.find('li').first();
  expect(firstRow.key()).toEqual(serviceCategories[0].id);
});

test('Snapshot renders', () => {
  const component = renderer.create(
    <ServiceCategorySectionList serviceCategories={serviceCategories} />
  );

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});
