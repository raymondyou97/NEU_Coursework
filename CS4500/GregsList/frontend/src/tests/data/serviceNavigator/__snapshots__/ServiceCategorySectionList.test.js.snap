// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot renders 1`] = `
<ul>
  <li
    style={
      Object {
        "listStyle": "none",
      }
    }
  >
    <p>
      <div>
        <h2
          id="123"
        >
          Home Improvements
        </h2>
        <div>
          <div>
            <div
              className="card-deck"
            >
              <div
                className="card"
                style={
                  Object {
                    "flex": "none",
                    "flexDirection": "column",
                    "flexWrap": "wrap",
                    "marginBottom": "15px",
                    "maxHeight": "100%",
                    "maxWidth": "50%",
                  }
                }
              >
                <img
                  className="card-img-top"
                  src="https://picsum.photos/300/200"
                />
                <div
                  className="card-body"
                >
                  <p
                    className="card-text"
                  >
                    <a
                      href="/ServiceProviderSearch/1"
                    >
                      Appliance Installation
                    </a>
                  </p>
                </div>
              </div>
              <div
                className="card"
                style={
                  Object {
                    "flex": "none",
                    "flexDirection": "column",
                    "flexWrap": "wrap",
                    "marginBottom": "15px",
                    "maxHeight": "100%",
                    "maxWidth": "50%",
                  }
                }
              >
                <img
                  className="card-img-top"
                  src="https://picsum.photos/300/200"
                />
                <div
                  className="card-body"
                >
                  <p
                    className="card-text"
                  >
                    <a
                      href="/ServiceProviderSearch/2"
                    >
                      Cabinetry
                    </a>
                  </p>
                </div>
              </div>
              <div
                className="card"
                style={
                  Object {
                    "flex": "none",
                    "flexDirection": "column",
                    "flexWrap": "wrap",
                    "marginBottom": "15px",
                    "maxHeight": "100%",
                    "maxWidth": "50%",
                  }
                }
              >
                <img
                  className="card-img-top"
                  src="https://picsum.photos/300/200"
                />
                <div
                  className="card-body"
                >
                  <p
                    className="card-text"
                  >
                    <a
                      href="/ServiceProviderSearch/3"
                    >
                      Door Installation
                    </a>
                  </p>
                </div>
              </div>
              <div
                className="card"
                style={
                  Object {
                    "flex": "none",
                    "flexDirection": "column",
                    "flexWrap": "wrap",
                    "marginBottom": "15px",
                    "maxHeight": "100%",
                    "maxWidth": "50%",
                  }
                }
              >
                <img
                  className="card-img-top"
                  src="https://picsum.photos/300/200"
                />
                <div
                  className="card-body"
                >
                  <p
                    className="card-text"
                  >
                    <a
                      href="/ServiceProviderSearch/4"
                    >
                      Concrete Removal
                    </a>
                  </p>
                </div>
              </div>
            </div>
            <div>
              <a
                href="/providers"
              >
                 
                Appliance Installation
              </a>
            </div>
            <div>
              <a
                href="/providers"
              >
                 
                Cabinetry
              </a>
            </div>
            <div>
              <a
                href="/providers"
              >
                 
                Door Installation
              </a>
            </div>
            <div>
              <a
                href="/providers"
              >
                 
                Concrete Removal
              </a>
            </div>
            <div>
              <a
                href="/providers"
              >
                 
                Dishwasher Installation
              </a>
            </div>
            <div>
              <a
                href="/providers"
              >
                 
                Finish Carpentry
              </a>
            </div>
            <div>
              <a
                href="/providers"
              >
                 
                Floor Polishing
              </a>
            </div>
            <div>
              <a
                href="/providers"
              >
                 
                Furniture Assembly
              </a>
            </div>
            <div>
              <a
                href="/providers"
              >
                 
                Generator Installation
              </a>
            </div>
            <div>
              <a
                href="/providers"
              >
                 
                Gardening
              </a>
            </div>
            <div>
              <a
                href="/providers"
              >
                 
                Home Organizing
              </a>
            </div>
            <div>
              <a
                href="/providers"
              >
                 
                Electrical and Wiring Repair
              </a>
            </div>
          </div>
        </div>
      </div>
    </p>
  </li>
  <li
    style={
      Object {
        "listStyle": "none",
      }
    }
  >
    <p>
      <div>
        <h2
          id="345"
        >
          Pets
        </h2>
        <div>
          <div>
            <div
              className="card-deck"
            >
              <div
                className="card"
                style={
                  Object {
                    "flex": "none",
                    "flexDirection": "column",
                    "flexWrap": "wrap",
                    "marginBottom": "15px",
                    "maxHeight": "100%",
                    "maxWidth": "50%",
                  }
                }
              >
                <img
                  className="card-img-top"
                  src="https://picsum.photos/300/200"
                />
                <div
                  className="card-body"
                >
                  <p
                    className="card-text"
                  >
                    <a
                      href="/ServiceProviderSearch/1"
                    >
                      Cat Grooming
                    </a>
                  </p>
                </div>
              </div>
              <div
                className="card"
                style={
                  Object {
                    "flex": "none",
                    "flexDirection": "column",
                    "flexWrap": "wrap",
                    "marginBottom": "15px",
                    "maxHeight": "100%",
                    "maxWidth": "50%",
                  }
                }
              >
                <img
                  className="card-img-top"
                  src="https://picsum.photos/300/200"
                />
                <div
                  className="card-body"
                >
                  <p
                    className="card-text"
                  >
                    <a
                      href="/ServiceProviderSearch/2"
                    >
                      Dog Training
                    </a>
                  </p>
                </div>
              </div>
              <div
                className="card"
                style={
                  Object {
                    "flex": "none",
                    "flexDirection": "column",
                    "flexWrap": "wrap",
                    "marginBottom": "15px",
                    "maxHeight": "100%",
                    "maxWidth": "50%",
                  }
                }
              >
                <img
                  className="card-img-top"
                  src="https://picsum.photos/300/200"
                />
                <div
                  className="card-body"
                >
                  <p
                    className="card-text"
                  >
                    <a
                      href="/ServiceProviderSearch/3"
                    >
                      Pet Boarding
                    </a>
                  </p>
                </div>
              </div>
              <div
                className="card"
                style={
                  Object {
                    "flex": "none",
                    "flexDirection": "column",
                    "flexWrap": "wrap",
                    "marginBottom": "15px",
                    "maxHeight": "100%",
                    "maxWidth": "50%",
                  }
                }
              >
                <img
                  className="card-img-top"
                  src="https://picsum.photos/300/200"
                />
                <div
                  className="card-body"
                >
                  <p
                    className="card-text"
                  >
                    <a
                      href="/ServiceProviderSearch/4"
                    >
                      Pet Sitting
                    </a>
                  </p>
                </div>
              </div>
            </div>
            <div>
              <a
                href="/providers"
              >
                 
                Cat Grooming
              </a>
            </div>
            <div>
              <a
                href="/providers"
              >
                 
                Dog Training
              </a>
            </div>
            <div>
              <a
                href="/providers"
              >
                 
                Pet Boarding
              </a>
            </div>
            <div>
              <a
                href="/providers"
              >
                 
                Pet Sitting
              </a>
            </div>
            <div>
              <a
                href="/providers"
              >
                 
                Dog Walking
              </a>
            </div>
            <div>
              <a
                href="/providers"
              >
                 
                Dog Grooming
              </a>
            </div>
            <div>
              <a
                href="/providers"
              >
                 
                Dog Daycare
              </a>
            </div>
            <div>
              <a
                href="/providers"
              >
                 
                Aquarium Services
              </a>
            </div>
            <div>
              <a
                href="/providers"
              >
                 
                Animal Training (Non-canine)
              </a>
            </div>
            <div>
              <a
                href="/providers"
              >
                 
                Behavior Modification (Non-canine)
              </a>
            </div>
          </div>
        </div>
      </div>
    </p>
  </li>
  <li
    style={
      Object {
        "listStyle": "none",
      }
    }
  >
    <p>
      <div>
        <h2
          id="234"
        >
          Wellness
        </h2>
        <div>
          <div>
            <div
              className="card-deck"
            />
          </div>
        </div>
      </div>
    </p>
  </li>
  <li
    style={
      Object {
        "listStyle": "none",
      }
    }
  >
    <p>
      <div>
        <h2
          id="456"
        >
          Business
        </h2>
        <div>
          <div>
            <div
              className="card-deck"
            />
          </div>
        </div>
      </div>
    </p>
  </li>
  <li
    style={
      Object {
        "listStyle": "none",
      }
    }
  >
    <p>
      <div>
        <h2
          id="567"
        >
          Events
        </h2>
        <div>
          <div>
            <div
              className="card-deck"
            />
          </div>
        </div>
      </div>
    </p>
  </li>
  <li
    style={
      Object {
        "listStyle": "none",
      }
    }
  >
    <p>
      <div>
        <h2
          id="678"
        >
          Crafts
        </h2>
        <div>
          <div>
            <div
              className="card-deck"
            />
          </div>
        </div>
      </div>
    </p>
  </li>
  <li
    style={
      Object {
        "listStyle": "none",
      }
    }
  >
    <p>
      <div>
        <h2
          id="789"
        >
          Design and Web
        </h2>
        <div>
          <div>
            <div
              className="card-deck"
            />
          </div>
        </div>
      </div>
    </p>
  </li>
  <li
    style={
      Object {
        "listStyle": "none",
      }
    }
  >
    <p>
      <div>
        <h2
          id="890"
        >
          Legal
        </h2>
        <div>
          <div>
            <div
              className="card-deck"
            />
          </div>
        </div>
      </div>
    </p>
  </li>
  <li
    style={
      Object {
        "listStyle": "none",
      }
    }
  >
    <p>
      <div>
        <h2
          id="987"
        >
          Personal
        </h2>
        <div>
          <div>
            <div
              className="card-deck"
            />
          </div>
        </div>
      </div>
    </p>
  </li>
  <li
    style={
      Object {
        "listStyle": "none",
      }
    }
  >
    <p>
      <div>
        <h2
          id="876"
        >
          Photography
        </h2>
        <div>
          <div>
            <div
              className="card-deck"
            />
          </div>
        </div>
      </div>
    </p>
  </li>
</ul>
`;
